src/algebra/order/floor.lean

```diff
@@ -35,10 +35,14 @@ We define the natural- and integer-valued floor and ceil functions on linearly o
 The index `₊` in the notations for `nat.floor` and `nat.ceil` is used in analogy to the notation
 for `nnnorm`.
 
-## TODO
+## Notes
 
-`linear_ordered_ring`/`linear_ordered_semiring` can be relaxed to `order_ring`/`order_semiring` in
-many lemmas.
+`linear_ordered_ring`/`linear_ordered_semiring` could be relaxed to `order_ring`/`order_semiring` in
+many lemmas, but this seems very unmathematical. `α` being a floor ring implies it being
+archimedean, in the sense that for all `a : α`, we have some `m n : ℤ` such that `m ≤ a` and
+`a ≤ n`. This rules out almost all non-linear rings. What's left looks like the semiring in
+the `canonically_ordered_comm_semiring_two_mul` counterexample, and does not seem to be of
+mathematical interest.
```

Comment:
In what sense does this rule out "almost all non-linear rings"? Why is it the case that any remaining ring looks like that counterexample? And what is the argument that these rings are not interesting?